qa/L0_metrics/test.sh

```diff
@@ -125,11 +127,26 @@ if [ "$SERVER_PID" == "0" ]; then
 fi
 
 num_iterations=10
+
+# Add "warm up" iteration because in some cases the GPU metrics collection
+# doesn't start immediately
+prev_energy=`curl -s localhost:8002/metrics | awk '/nv_energy_consumption{/ {print $2}'`
+for (( i = 0; i < $num_iterations; ++i )); do
+  sleep $WAIT_INTERVAL_SECS
+  current_energy=`curl -s localhost:8002/metrics | awk '/nv_energy_consumption{/ {print $2}'`
+  if [ $current_energy != $prev_energy ]; then
+    echo -e "\n***\n*** Detected changing metrics, warmup completed.\n***"
+    break
+  fi
+  prev_energy=$current_energy
+done
+
 prev_energy=`curl -s localhost:8002/metrics | awk '/nv_energy_consumption{/ {print $2}'`
 for (( i = 0; i < $num_iterations; ++i )); do
-  sleep $METRICS_INTERVAL_SECS
+  sleep $WAIT_INTERVAL_SECS
   current_energy=`curl -s localhost:8002/metrics | awk '/nv_energy_consumption{/ {print $2}'`
   if [ $current_energy == $prev_energy ]; then
+    cat $SERVER_LOG
     echo "Metrics were not updated in interval of ${METRICS_INTERVAL_MS} seconds"
```

Comment:
Not part of your changes, but can you change `interval of ${METRICS_INTERVAL_MS} seconds"` to say `milliseconds"` here instead? It's misleading in log output when it says "metrics not updated for 500 seconds"